Ark.Tools.FtpClient.SftpClient/SftpClientConnection.cs

```diff
@@ -87,19 +108,117 @@ public override async Task UploadFileAsync(string path, byte[] content, Cancella
 
         #region private helpers
 
-        private Renci.SshNet.SftpClient _getSFtpClient()
+        private ConnectionInfo _getConnectionInfo()
+        {
+            var connrectionInfo = new ConnectionInfo(Uri == null ? Host : Uri.Host, Uri == null ? Port : Uri.Port, Credentials.UserName, new PasswordAuthenticationMethod(Credentials.UserName, Credentials.Password))
+            {
+                Timeout = TimeSpan.FromMinutes(5),
+                RetryAttempts = 2
+            };
+
+            return connrectionInfo;
+        }
+
+        private Renci.SshNet.SftpClient _getSFtpClientWithCertificate()
         {
-            var connInfo = new ConnectionInfo(Uri == null ? Host : Uri.Host, Uri == null ? Port : Uri.Port, Credentials.UserName, new PasswordAuthenticationMethod(Credentials.UserName, Credentials.Password));
-            connInfo.Timeout = TimeSpan.FromMinutes(5);
-            connInfo.RetryAttempts = 2;
-           
-            return new Renci.SshNet.SftpClient(connInfo)
+
+#if NETSTANDARD2_0
+            throw new NotSupportedException($"ClientCertificate does not support X509 Certificate in NETCORE2.0");
+#endif
+            var connInfo = _getConnectionInfo();
+
+            using var cert = _ftpConfig.ClientCertificate;
```

Comment:
nope, you're disposing the cert. FtpConfig owns the Cert, not _getSFtpClientWithCertificate